src/cmd/services/m3coordinator/handler/database/create.go

```diff
@@ -0,0 +1,251 @@
+		return
+	}
+
+	nsRegistry, err := h.namespaceAddHandler.Add(namespaceRequest)
+	if err != nil {
+		logger.Error("unable to add namespace", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	initPlacement, err := h.placementInitHandler.Init(placementRequest)
+	if err != nil {
+		// Attempt to delete the namespace that was just created to maintain idempotency
+		err = h.namespaceDeleteHandler.Delete(namespaceRequest.Name)
+		if err != nil {
+			logger.Error("unable to delete namespace we just added", zap.Any("error", err))
+			handler.Error(w, err, http.StatusInternalServerError)
+			return
+		}
+
+		logger.Error("unable to add namespace", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	placementProto, err := initPlacement.Proto()
+	if err != nil {
+		logger.Error("unable to get placement protobuf", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	resp := &admin.DatabaseCreateResponse{
+		Namespace: &admin.NamespaceGetResponse{
+			Registry: &nsRegistry,
+		},
+		Placement: &admin.PlacementGetResponse{
+			Placement: placementProto,
+		},
+	}
+
+	handler.WriteProtoMsgJSONResponse(w, resp, logger)
+}
+
+func (h *createHandler) parseRequest(r *http.Request) (*admin.NamespaceAddRequest, *admin.PlacementInitRequest, *handler.ParseError) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, nil, handler.NewParseError(err, http.StatusBadRequest)
+
+	}
+	defer r.Body.Close()
+
+	dbCreateReq := new(admin.DatabaseCreateRequest)
+	if err := json.Unmarshal(body, dbCreateReq); err != nil {
+		return nil, nil, handler.NewParseError(err, http.StatusBadRequest)
+	}
+
+	if dbCreateReq.Type != dbTypeLocal {
+		return nil, nil, handler.NewParseError(errInvalidDBType, http.StatusBadRequest)
+	}
+
+	if util.HasEmptyString(dbCreateReq.NamespaceName, dbCreateReq.Type) {
+		return nil, nil, handler.NewParseError(errMissingRequiredField, http.StatusBadRequest)
+	}
+
+	namespaceAddRequest := defaultedNamespaceAddRequest(dbCreateReq.NamespaceName, dbCreateReq.Type)
+	placementInitRequest := defaultedPlacementInitRequest(dbCreateReq)
+
+	return namespaceAddRequest, placementInitRequest, nil
+}
+
+func defaultedNamespaceAddRequest(namespaceName, validDBType string) *admin.NamespaceAddRequest {
+	var (
+		bootstrapEnabled  bool
+		flushEnabled      bool
+		writesToCommitLog bool
+		cleanupEnabled    bool
+		repairEnabled     bool
+		snapshotEnabled   bool
+
+		retentionPeriodNanos                     int64
+		retentionBlockSizeNanos                  int64
+		bufferFutureNanos                        int64
+		bufferPastNanos                          int64
+		blockDataExpiry                          bool
+		blockDataExpiryAfterNotAccessPeriodNanos int64
+
+		indexEnabled        bool
+		indexBlockSizeNanos int64
+	)
+
+	switch validDBType {
+	case dbTypeLocal:
+		bootstrapEnabled = true
```

Comment:
Are these sane defaults?

Reply:
This looks good for a local DB, good stuff.